ultralytics/data/annotator.py

```diff
@@ -64,3 +71,337 @@ def auto_annotate(
                     if s.any():
                         segment = map(str, s.reshape(-1).tolist())
                         f.write(f"{class_ids[i]} " + " ".join(segment) + "\n")
+
+
+class AutoAnnotator:
+    """
+    A class for automatic annotation of images using YOLO or Florence2 models.
+
+    This class handles model selection, object detection, and label generation in YOLO format.
+    It supports both YOLO-based and Florence2-based annotation pipelines, including visual output.
+
+    Attributes:
+        Y0LO_MODEL (bool): Flag indicating if the annotation uses a YOLO model.
+        model (YOLO | Florence2): Loaded model instance for annotation.
+        processor (Florence2Processor): Used only for Florence2-based annotation.
+        names (dict): Class names for YOLO-based models.
+
+    Methods:
+        annotate: Performs annotation on a dataset of images or videos.
+        convert_to_yolo: Converts bounding box coordinates to YOLO normalized format.
+        create_classes_txt: Saves class names into a `classes.txt` file.
+    """
+
+    def __init__(self, model=None, device="", variant="base"):
+        """
+        Initializes the AutoAnnotator with either a YOLO or Florence2 model.
+
+        Args:
+            model (str, optional): Model name or path. Defaults to "yolo11n.pt".
+            device (str): Device to use for inference (e.g., 'cpu', 'cuda:0').
+            variant (str): Florence2 model variant if using Florence2 (e.g., "base", "large-ft").
+        """
+        self.YOLO_MODEL = True  # Assume YOLO by default
+
+        if model is None:
+            LOGGER.warning("⚠️ No model provided. Using default: yolo11n.pt")
+            self.model = "yolo11n.pt"
+        else:
+            model_str = str(model).casefold()
+
+            if model_str.endswith(".pt") and model_str not in ("florence2.pt", "florence2"):
+                self.model = YOLO(model)
+            elif model_str in ("florence2.pt", "florence2"):
+                self.model = Florence2(device=device, variant=variant)
+                self.YOLO_MODEL = False
+            else:
+                LOGGER.warning(
+                    f"⚠️ Unsupported model '{model}'. Using default: yolo11n.pt.\n"
+                    f"✅ Supported models:\n"
+                    f"   - YOLO: yolov8n.pt, yolov11n.pt, yolov10.pt, yolo11n.pt, yolo12n.pt\n"
```

Comment:
Is there a yolov10.pt checkpoint file?

Also this makes it look like only nano-sized models are supported for YOLO. Please update the copy here to explain the list of supported models more clearly.